framework/include/base/MooseApp.h

```diff
@@ -109,14 +109,13 @@ class MooseApp : public ConsoleStreamInterface,
    */
   virtual std::string getPrintableName() const { return "Application"; }
 
-  virtual std::string appBinaryName() const
-  {
-    auto name = Moose::getExecutableName();
-    name = name.substr(0, name.find_last_of("-"));
-    if (name.find_first_of("/") != std::string::npos)
-      name = name.substr(name.find_first_of("/") + 1, std::string::npos);
-    return name;
-  }
+  /**
+   * @return The app's binary name
+   *
+   * If \p real, then return the "real" binary name in the event that
+   * the binary is a symlink
+   */
+  std::string appBinaryName(const bool real = false) const;
```

Comment:
You dropped virtual. Are we sure we never need to override this? What about SAM? Do we have any special cases?

Reply:
I will approve after talking with you and verifying that this is intentional.